src/middleware.ts

```diff
@@ -258,36 +263,63 @@ export const persist = <S extends State>(
     void setItem()
   }
 
+  const makeThenable = (value: any) => {
+    if (value instanceof Promise) {
+      return value
+    } else {
+      return {
+        then: (callback) => callback(value),
```

Comment:
```suggestion
        then: (callback) => makeThenable(callback(value)),
```
Not strictly necessary, but maybe a good idea. I had a bug where I forgot to manually wrap the result of `then()` with `makeThenable()`. Also it would resemble real promises a more.

Reply:
good idea.